rust/src/push/evaluator.rs

```diff
@@ -325,32 +370,12 @@ impl PushRuleEvaluator {
     /// Evaluates a `event_match` condition.
     fn match_event_match(
         &self,
-        event_match: &EventMatchCondition,
-        user_id: Option<&str>,
+        flattened_event: &BTreeMap<String, JsonValue>,
+        key: &str,
+        pattern: &str,
```

Comment:
Or should these be:

```suggestion
        key: Cow<str>,
        pattern: Cow<str>,
```

Reply:
I think `&str` is correct---you only ever read these string slices, and `Cow<'a, T>` can be transformed into an `&'a T` at essentially zero cost I think?